smart-doc/README.md

```diff
@@ -0,0 +1,176 @@
+
+* Assembly/article files:
+
+      articles/<NAME>.asm.xml
+
+* Concept, reference, task, glue files:
+
+      concepts/<NAME>.xml
+      references/<NAME>.xml
+      tasks/<NAME>.xml
+      glues/<NAME>.xml
+
+* Common files (entities, gfdl, etc):
+
+      common/<NAME>.xml
+
+* Any topic's name consists of the following elements that are explained below:
+
+      <NAME> := <TOPIC>[-<SUBTOPIC]-<SNIPPET-TOPIC>-<TITLE>
+
+### DC file naming conventions
+
+The DC file name must match the assembly file's root ID.
+
+### File naming conventions
+
+Always start with the overarching topic. In doing so, ensure that articles belonging to the same topic can be easily spotted by someone looking for existing articles. If necessary, add a subtopic (append with `_`):
+```
+autoyast
+systemd
+systemd-timer
+network
+```
+
+The second component for concept, reference, task, glue should be the topic of the snippet. For tasks, use a verb in gerund. For descriptive topics (concepts, references) opt for noun-based:
+
+```
+references/autoyast-partitioner-syntax.xml
+concepts/autoyast-control-file.xml
+tasks/systemd-timer-creating.xml
+tasks/systemd-timer-troubleshooting.xml
+concepts/network-ipv6-vs-ipv4.xml
+glues/systemd-timer-intro.xml
+glues/systemd-timer-whats-next.xml
+glues/systemd-timer-more-info.xml
+```
+
+The second component of the article should be an abbreviated title. As our articles are intended to address specific problems and tasks, use verbs in gerund here as well:
+
+```
+articles/systemd-timer-working-with.asm.xml
+articles/autoyast-understanding-config-file.asm.xml
+```
+
+Distinguish article files from standard XML files by adding the `.asm` suffix.
+
+
+### ID naming conventions
+
+Working with assemblies, you need to distinguish between two types of IDs:
+
+* `external IDs`: These are visible to the world.
+* `internal IDs`: Internal references inside the assembly used to organize the content snippets.
+
+External IDs are built in a similar fashion as file names. As with file names, never include the 
+topic type in your ID.
+
+```
+<NAME> := <TOPIC>[-<SUBTOPIC]-<SNIPPET-TOPIC>-<TITLE>
+```
+
+To build an internal ID, prefix the external ID with an underscore (`_`).
+
+
+### Image naming conventions
+
+Make sure your images can be associated with the topic they belong to. Use a similar naming scheme 
+for images that you use for topic files.
+
+
+```
+<NAME> := <TOPIC>[-<SUBTOPIC]-<SNIPPET-TOPIC>-<TITLE>
+```
+## Version info
+
+We maintain a changelog for each article. Add a version to the changelog whenever there is a 
+*release* of your article, i.e. whenever you ask for a rebuild and repost of said article. Do not 
+artificially bloat the changelog by entering every single commit. Also, provide concise and 
+meaningful version information that the reader benefits from. This information might one day end up 
+being sent out to readers as update notifications on content they subscribed to.
```

Comment:
I think the term "changelog" and this paragraph needs some additions: :wink: 

1. Perhaps you could mention revision histories as a synonym here?
2. The paragraph doesn't mention _where_ to put the information. Maybe add a brief sentence ("... add it to the assembly...")
3. I would also mention somewhere that changelog means adding/using `<revhistory>` element.

It doesn't have to be many details, but I think it would be helpful to have this mental connection that changelog = `<revhistory>` somewhere.

Reply:
@tomschr: I just copied this README from SUSE/doc-modular... I will check with @janajaeger where and when to integrate the changes you suggested (if we keep the file in doc-kit, it should be maintained here of course).

Reply:
Good idea - this was obviously written before we had explicitly defined how to maintain the changelog. I have integrated your feedback in https://github.com/openSUSE/doc-kit/pull/78.